python/educational-core/student/src/com/jetbrains/edu/learning/stepic/StepicConnectorGet.java

```diff
@@ -0,0 +1,327 @@
+    final StepicWrappers.SectionContainer
+      sectionContainer = getFromStepic(EduStepicNames.SECTIONS + String.valueOf(sectionId), StepicWrappers.SectionContainer.class);
+    List<Integer> unitIds = sectionContainer.sections.get(0).units;
+
+    StepicWrappers.UnitContainer
+      unitContainer = getFromStepic(EduStepicNames.UNITS + "/" + getIdQuery(unitIds), StepicWrappers.UnitContainer.class);
+    List<Integer> lessonsIds = new ArrayList<>();
+    unitContainer.units.forEach(x -> lessonsIds.add(x.lesson));
+    StepicWrappers.LessonContainer
+      lessonContainer = getFromStepic(EduStepicNames.LESSONS + getIdQuery(lessonsIds), StepicWrappers.LessonContainer.class);
+
+    String sectionName = sectionContainer.sections.get(0).title;
+    final List<Lesson> lessons = new ArrayList<Lesson>();
+    for (Lesson lesson : lessonContainer.lessons) {
+      lesson.setName(sectionName + EduNames.SEPARATOR + lesson.getName());
+      //      LOG.info("set lesson name " + lesson.getName());
+      createTasks(lesson, lesson.steps);
+      if (!lesson.taskList.isEmpty()) {
+        lessons.add(lesson);
+      }
+    }
+    return lessons;
+  }
+
+  private static void createTasks(Lesson lesson, List<Integer> stepicIds) throws IOException {
+    final StepicWrappers.StepContainer stepContainer = getSteps(stepicIds);
+    List<StepicWrappers.Step> steps = new ArrayList<>();
+    stepContainer.steps.forEach(x -> steps.add(x.block));
+    int i = 0;
+    for (StepicWrappers.Step step : steps) {
+      if (supported(step.name)) {
+        final Task task = new Task();
+        task.setStepicId(stepicIds.get(i++));
+
+        switch (step.name) {
+          case (CODE_PREFIX):
+            createCodeTask(task, step);
+            break;
+          case (PYCHARM_PREFIX):
+            createPyCharmTask(task, step);
+            break;
+        }
+        lesson.taskList.add(task);
+      }
+    }
+  }
+
+  private static boolean supported(String name) {
+    return CODE_PREFIX.equals(name) || PYCHARM_PREFIX.equals(name);
+  }
+
+  private static void createPyCharmTask(Task task, StepicWrappers.Step step) {
+    task.setName(step.options != null ? step.options.title : PYCHARM_PREFIX);
+    task.setText(step.text);
+    for (StepicWrappers.TestFileWrapper wrapper : step.options.test) {
+      task.addTestsTexts(wrapper.name, wrapper.text);
+    }
+
+    task.taskFiles = new HashMap<String, TaskFile>();      // TODO: it looks like we don't need taskFiles as map anymore
+    if (step.options.files != null) {
+      for (TaskFile taskFile : step.options.files) {
+        task.taskFiles.put(taskFile.name, taskFile);
+      }
+    }
+  }
+
+  private static void createCodeTask(Task task, StepicWrappers.Step step) {
+    task.setName("step" + task.getStepicId());
+    if (step.options.samples != null) {
+      final StringBuilder builder = new StringBuilder();
+      for (List<String> sample : step.options.samples) {
+        if (sample.size() == 2) {
```

Comment:
I saw exactly the same copy-pasted code in this pull request already in different class